gtsfm/loader/yfcc_imb_loader.py

```diff
@@ -0,0 +1,190 @@
+            index: the index to fetch.
+
+        Raises:
+            IndexError: if an out-of-bounds image index is requested.
+
+        Returns:
+            Image: the image at the query index.
+        """
+        if index < 0 or index > self.__len__():
+            raise IndexError("Image index is invalid")
+
+        image_name = self._image_names[index]
+
+        file_name = osp.join(
+            self._folder, "images", "{}.jpg".format(image_name)
+        )
+
+        return io_utils.load_image(file_name)
+
+    def get_camera_intrinsics(self, index: int) -> Optional[Cal3Bundler]:
+        """Get the camera intrinsics at the given index.
+
+        Args:
+            the index to fetch.
+
+        Returns:
+            intrinsics for the given camera.
+        """
+        if index < 0 or index > self.__len__():
+            raise IndexError("Image index is invalid")
+
+        return self._cameras[index].calibration()
+
+    def get_camera_pose(self, index: int) -> Optional[Pose3]:
+        """Get the camera pose (in world coordinates) at the given index.
+
+        Args:
+            index: the index to fetch.
+
+        Returns:
+            the camera pose w_P_index.
+        """
+        if index < 0 or index > self.__len__():
+            raise IndexError("Image index is invalid")
+
+        return self._cameras[index].pose()
+
+    def validate_pair(self, idx1: int, idx2: int) -> bool:
+        """Checks if (idx1, idx2) is a valid pair.
+
+        Args:
+            idx1: first index of the pair.
+            idx2: second index of the pair.
+
+        Returns:
+            validation result.
+        """
+        return (idx1, idx2) in self._image_pairs
+
+    def __read_colmap_model(self) -> List[PinholeCameraCal3Bundler]:
+        """Read camera parameters from the COLMAP optimized result provided by
+        the IMB.
+
+        Returns:
+            list of all cameras.
+        """
+
+        # read the COLMAP optimized params for ground truth
+        camera_params, image_params, _ = colmap_read_write_model.read_model(
+            path=osp.join(self._folder, "sparse"), ext=".bin"
+        )
+
+        # map our indices to param indices
+        indices_map = {}
+        for k, v in image_params.items():
+            filename = osp.splitext(v.name)[0]
+
+            if filename in self._name_to_idx_map:
+                indices_map[self._name_to_idx_map[filename]] = k
+
+        cameras = []
+
+        # load params for all image names
+        for idx in range(len(self)):
+            quaternions = image_params[indices_map[idx]].qvec
+
+            rot = Rot3(
+                w=quaternions[0],
+                x=quaternions[1],
+                y=quaternions[2],
+                z=quaternions[3],
+            )
+
+            cTw = Pose3(rot, image_params[indices_map[idx]].tvec)
+
+            K_params = camera_params[indices_map[idx]].params
+            # TODO: cannot fit two focal lengths k_params[0] and k_params[1]
```

Comment:
thanks for the PR, Ayush. In the IMB ground truth, are there any cameras w/ nonsquare pixels? (fx != fy)

Reply:
Yup. the images are non-square

Reply:
What about the pixels themselves (not the images)?

If the pixels are square, then fx=fy, so the TODO could be removed. But pixels are usually square